GVFS/GVFS/CommandLine/UpgradeVerb.cs

```diff
@@ -102,24 +106,25 @@ private bool TryRunProductUpgrade()
                 return false;
             }
 
-            if (!this.TryLoadUpgradeRing(out ring, out error))
-            {
-                this.tracer.RelatedError($"{nameof(this.TryRunProductUpgrade)}: Could not load upgrade ring. {error}");
-                this.ReportInfoToConsole(GVFSConstants.UpgradeVerbMessages.InvalidRingConsoleAlert);
-                this.Output.WriteLine(errorOutputFormat, error);
-                return false;
-            }
-
-            if (ring == ProductUpgrader.RingType.None || ring == ProductUpgrader.RingType.NoConfig)
+            bool isError;
+            string consoleMessage;
+            if (!this.upgrader.CanRunUsingCurrentConfig(out isError, out consoleMessage, out error))

```

Comment:
I'm having a very hard time parsing this. `CanRunUsingCurrentConfig` returns bool, which I would expect to be whether or not we can run. But we also have an out param `isError`. What's an example of a time where we cannot run but it is not an error? Can we simplify this?

Reply:
Couple times this can happen - when upgrade ring is `none` or user has not configured the ring yet. In both cases, upgrade cannot run but its not an error.